hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -315,16 +324,31 @@ protected void writeInsertRecord(HoodieRecord<T> newRecord) throws IOException {
 
   protected void writeInsertRecord(HoodieRecord<T> newRecord, Schema schema, Properties prop)
       throws IOException {
-    if (writeRecord(newRecord, Option.of(newRecord), schema, prop, HoodieOperation.isDelete(newRecord.getOperation()))) {
+    if (writeRecord(newRecord, Option.empty(), Option.of(newRecord), schema, prop, HoodieOperation.isDelete(newRecord.getOperation()))) {
```

Comment:
can we just pass null instead of `Option.empty()` if the old record is not there.

Reply:
That could lead to NPE if somebody uses the old record directly. This could be safer bet.